### PR TITLE
Base fee and chain capacity tracking.

### PR DESCRIPTION
**Pull request by frrist**
### What
- Implement Message Gas Economy model for tracking Chain Capacity Usage and Base fee.
- Fix what I believe to be a bug in the migration logic by including a null migration at version 0 (I _think_ this might relate to #89 but am not sure).

### CODE REVIEW

File: tasks/message/message.go [hunk 1]
```diff
@@ -173,54 +184,75 @@ func (p *MessageProcessor) processTipSet(ctx context.Context, ts *types.TipSet)
...
 		for _, message := range vmm {
-			bmsgs = append(bmsgs, &messagemodel.BlockMessage{
+			// record which blocks had which messages
+			bmsgModels = append(bmsgModels, &messagemodel.BlockMessage{
 				Block:   blk.String(),
 				Message: message.Cid().String(),
 			})
 
-			// so we don't create duplicate message models.
+			totalUniqGasLimit += message.GasLimit
```
Comment (willscott) [inline tasks/message/message.go:238]: is it useful to double count limit for messages in both bls and secpk?

**Review by willscott — APPROVED** [review]

**Comment by iand** [comment]
Rebased onto master and changed the migration number to 10.

**Review by iand — APPROVED** [review]
LGTM

**Comment by iand** [comment]
I'm curious what the problem was that required a '0' migration

**Comment by frrist** [comment]
> I'm curious what the problem was that required a '0' migration

When the database is first initialized we run this query (https://github.com/go-pg/migrations/blob/v8/collection.go#L620) against it to get its current version. The gopg_migration table is empty so 0 is returned (https://github.com/go-pg/migrations/blob/v8/collection.go#L624) making the dbVersion at startup 0 (https://github.com/filecoin-project/sentinel-visor/blob/master/storage/migrate.go#L39)

**Pull request merged**